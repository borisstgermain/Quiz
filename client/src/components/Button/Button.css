.button {
  display: block;

  box-sizing: border-box;
  padding-right: 2%;
  padding-left: 2%;

  color: #421b75;
  border: none;
  background-color: #ffe258;

  font-size: 18px;
  line-height: 2.6em;
}

.button--full-width {
  width: 100%;
}
