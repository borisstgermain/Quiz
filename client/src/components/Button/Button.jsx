import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.css';


const propTypes = {
  label: PropTypes.string.isRequired,
  onClickHandler: PropTypes.func.isRequired,
  fullWidth: PropTypes.boolean,
};


export default function Button(props) {
  const { label, onClickHandler, fullWidth } = props;
  const classNames = `${styles.button}  ${fullWidth ? styles['button--full-width'] : ''}`;

  return (
    <button
      className={classNames}
      onClick={onClickHandler}
    >
      {label}
    </button>
  );
}

Button.propTypes = propTypes;
