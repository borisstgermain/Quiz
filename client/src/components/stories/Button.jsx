import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Button from '../Button';
import '../Button/Button.css';

storiesOf('Button', module)
  .add('Simple button', () => (
    <Button
      label={'Button'}
      onClickHandler={action('clicked')}
    />
  ))
  .add('Full-Width Button', () => (
    <Button
      label={'Full-width Button'}
      onClickHandler={action('clicked')}
      fullWidth
    />
  ));
