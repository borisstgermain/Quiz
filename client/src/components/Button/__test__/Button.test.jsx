import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';


describe('Button', () => {
  describe('Smoke tests', () => {
    test('Render without crashing', () => {
      shallow(<Button />);
    });
  });


  describe('Component test', () => {
    let props;
    let wrapper;

    beforeEach(() => {
      props = {
        label: 'Button <+>',
        fullWidth: true,
        onClickHandler: jest.fn(),
      };
      wrapper = shallow(<Button {...props} />);
    });

    test('Correct props', () => {
      expect(wrapper.instance().props.label).toEqual(props.label);
      expect(wrapper.instance().props.fullWidth).toEqual(props.fullWidth);
      expect(wrapper.instance().props.onClickHandler).toEqual(props.onClickHandler);
    });

    test('onClick handler', () => {
      const fn = wrapper.instance().props.onClickHandler;
      wrapper.find('button').simulate('click');

      expect(fn).toHaveBeenCalled();
    });

    test('Snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
