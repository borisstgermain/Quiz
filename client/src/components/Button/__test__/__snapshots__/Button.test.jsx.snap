// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component test Snapshot 1`] = `
<button
  className="button  button--full-width"
  onClick={[Function]}
>
  Button &lt;+&gt;
</button>
`;
